doc(sc-kafka-events): enhance EN and add FR version CTOR-1118 (#4362)
centreon/centreon-documentation: versioned_docs/version-24.10/integrations/data-analytics/sc-kafka-events.md

Author: ```suggestion
title: Kafka Events
```

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> The Kafka Events stream connector allows you to send data from Centreon to Kafka instances.

Reviewer: same as FR

| ------ |-----------|--------------------------------------------|-------------------------------------------|
| string | logfile   | The file in which logs are written         | /var/log/centreon-broker/kafka-events.log |
| number | log_level | Logging level from 1 (errors) to 3 (debug) | 1                                         |

7. Use the stream connector's optional parameters to [filter or adapt the data you want Centreon to send to Kafka](#filtering-or-adapting-the-data-you-want-to-send-to-kafka).

In addition to parameters from stream connectors, there is a handful of parameters available thanks to the librdkafka library. 
They are all documented in the librdkafka **[official documentation](https://github.com/edenhill/librdkafka/blob/v0.11.4/CONFIGURATION.md)**. 
To use them you just need to **add** the **_sc_kafka_ prefix**.

With that in mind, the **sasl.mechanism** parameter becomes **_sc_kafka_sasl.mechanism** in your broker configuration.

8. [Deploy the configuration](../../monitoring/monitoring-servers/deploying-a-configuration.md).

9. Restart **centengine** on all pollers:

   ```shell
   systemctl restart centengine
   ```

> Kafka should now receive data from Centreon. To test if it is working, see [Curl commands: testing the stream connector](#curl-commands-testing-the-stream-connector).

Reviewer: ```suggestion
Kafka should now receive data from Centreon. To test if it is working, see [Curl commands: testing the stream connector](#curl-commands--testing-the-stream-connector).
```

</TabItem>
</Tabs>

## Configuring your Kafka server

You may need to configure your Kafka server so that it can receive data from Centreon. Please refer to Kafka's documentation.
Make sure Kafka is able to receive data sent by Centreon: flows must not be blocked by Kafka's configuration or by a security device.

## Configuring the stream connector in Centreon

1. On your central server, go to **Configuration > Pollers > Broker configuration**.
2. Click on **central-broker-master** (or the appropriate broker configuration if it is a poller or a remote server that will send events).
3. On the **Output** tab, select **Generic - Stream connector** from the list and then click **Add**. A new output appears in the list.
4. Fill in the fields as follows:

| Field           | Value                                                 |
| --------------- |-------------------------------------------------------|
| Name            | Kafka events                                          |
| Path            | /usr/share/centreon-broker/lua/kafka-events-apiv2.lua |
| Filter category | Neb,Bam                                               |

> 5. To enable Centreon to connect to your Kafka equipment, fill in the following mandatory parameters. The fields for 

Reviewer: ```suggestion
5. To enable Centreon to connect to your Kafka device, fill in the following mandatory parameters. The fields for 
```

the first entry are already present. Click on the **+Add a new entry** link located below the **Filter category** table to add another one.

| Type   | Name    | Value explanation                                             | Value exemple                               |
| ------ | ------- |---------------------------------------------------------------|---------------------------------------------|
| string | topic   | The topic in which events are going to be written             | Monitoring                                  |

> | string | brokers | Coma separeted list of brokers that are ready to receive data | broker_address1:port1,broker_address2:port2 |

Reviewer: ```suggestion
| string | brokers | Comma-separated list of brokers that are ready to receive data | broker_address1:port1,broker_address2:port2 |
```

6. Fill in any optional parameters you want (using the **+Add a new entry** link):

| Type   | Name      | Value explanation                          | Default value                             |
| ------ |-----------|--------------------------------------------|-------------------------------------------|
| string | logfile   | The file in which logs are written         | /var/log/centreon-broker/kafka-events.log |
| number | log_level | Logging level from 1 (errors) to 3 (debug) | 1                                         |

7. Use the stream connector's optional parameters to [filter or adapt the data you want Centreon to send to Kafka](#filtering-or-adapting-the-data-you-want-to-send-to-kafka).

> In addition to parameters from stream connectors, there is a handfull of parameters available thanks to the librdkafka library. 

Reviewer: ```suggestion
In addition to parameters from stream connectors, there is a handful of parameters available thanks to the librdkafka library. 
```

They are all documented in the librdkafka **[official documentation](https://github.com/edenhill/librdkafka/blob/v0.11.4/CONFIGURATION.md)**. 
To use them you just need to **add** the **_sc_kafka_ prefix**.

> With that in mind, the parameter **sasl.mechanism** becomes **_sc_kafka_sasl.mechanism** in your broker configuration.

Reviewer: ```suggestion
With that in mind, the **sasl.mechanism** parameter becomes **_sc_kafka_sasl.mechanism** in your broker configuration.
```

> > El7 and El8 repos grant access to an old librdkafka library version.

Reviewer: same as FR

> > El7 and El8 repos grant access to an old librdkafka library version.

Author: ```suggestion
```

8. [Deploy the configuration](../../monitoring/monitoring-servers/deploying-a-configuration.md).

9. Restart **centengine** on all pollers:

   ```shell
   systemctl restart centengine
   ```

Kafka should now receive data from Centreon. To test if it is working, see [Curl commands: testing the stream connector](#curl-commands-testing-the-stream-connector).

### Filtering or adapting the data you want to send to Kafka

> All stream connectors have a set of [optional parameters](https://github.com/centreon/centreon-stream-connector-scripts/blob/master/modules/docs/sc_param.md#default-parameters), that allow you to filter the data you will send to your Kafka equipment, to reformat the data, to define a proxy...

Reviewer: ```suggestion
All stream connectors have a set of [optional parameters](https://github.com/centreon/centreon-stream-connector-scripts/blob/master/modules/docs/sc_param.md#default-parameters), that allow you to filter the data you will send to your Kafka device, to reformat the data, to define a proxy...
```

### Custom event format

This stream connector allows you to change the format of the event to suit your needs. Only the **event** part of the json is customisable. 
It also allows you to handle event types that are not handled by default such as **ba_status events**.

In order to use this feature you need to configure a json event format file and add a new stream connector parameter.

| Type   | Name        | Value                                         |
| ------ | ----------- |-----------------------------------------------|
| string | format_file | /etc/centreon-broker/kafka-events-format.json |

> > The event format configuration file must be readable by the centreon-broker user.

Reviewer: ```suggestion
> The event format configuration file must be readable by the **centreon-broker** user.
```

To learn more about custom event formats and templating files, visit **[this page](https://github.com/centreon/centreon-stream-connector-scripts/blob/master/modules/docs/templating.md#templating-documentation)**.

## Testing the stream connector

> Sending data to Kafka can be quite complicated because of all the involved parameters (either from the stream connector itself or the kafka library).

Reviewer: ```suggestion
Sending data to Kafka can be quite complicated because of all the involved parameters (either from the stream connector itself or the Kafka library).
```

There are already configuration set up as examples to guide you.

If it doesn't work, you should have an error message like below (with the appropriate error message). 

> It is strongly advised to have access to kafka to check if a message is sent from the test script.

Reviewer: ```suggestion
It is strongly advised to have access to Kafka to check if a message is sent from the test script.
```
